### PR TITLE
Gradebook Test Plan

### PR DESCRIPTION
**Pull request by nsprenkle**
**TL;DR -** Add a first-pass at a test plan.

JIRA: [EDUCATOR-5712](https://openedx.atlassian.net/browse/EDUCATOR-5712)

**What changed?**

- Adds `test-setup.md` for basic course/feature/track setup needed for testing Gradebook.
- Enumerates the basic workflows and provides steps to test/verify.
- Adds reminder to the _Developer Checklist_ to update documentation.

**Developer Checklist**
- [x] Test suites passing
- [ ] Documentation and test plan updated, if applicable
- [ ] Received code-owner approving review
- [ ] Bumped version number [package.json](../package.json)

**Testing Instructions**

- Verify workflows and setup instructions. I suggest viewing the "Rich Diffs" for files so you can see what they will look like in rendered Markdown.
- Critique any parts that are unclear.

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [x] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @edx/masters-devs-gta

### CODE REVIEW

**Comment by mattcarter** [comment]
This is a good start.  Any reason to keep it as `draft`?

**Comment by nsprenkle** [comment]
> This is a good start. Any reason to keep it as `draft`?

@mattcarter, not really. I think I've put in all the starting pieces I need. Will move to Ready for review.

**Review by mattcarter — APPROVED** [review]
This looks like a good kick-off for me. Approved 👍

**Review by jnlapierre — APPROVED** [review]
This document is gold and an excellent onboarding exercise in addition to smoke test.
